src/protagonist/DLCS.AWS/SNS/ITopicPublisher.cs

```diff
@@ -0,0 +1,12 @@
+﻿namespace DLCS.AWS.SNS;
+
+public interface ITopicPublisher
+{
+    /// <summary>
+    /// Asynchronously publishes a message to an SNS topic
+    /// </summary>
+    /// <param name="messageContents">The contents of the message</param>
+    ///  /// <param name="subscribedQueueType">The type of subscribed queue to publish a message to</param>
+    /// <param name="cancellationToken">A cancellation token</param>
+    public Task<bool> PublishToTopicAsync(string messageContents, SubscribedQueueType subscribedQueueType, CancellationToken cancellationToken);
```

Comment:
This should take a variables that specifies which queue is to be published to, or should have name updated to more accurately show what it does. Something like:

```cs
public Task<bool> PublishToAssetModifiedTopicAsync(string messageContents, SubscribedQueueType subscribedQueueType, CancellationToken cancellationToken);

public Task<bool> PublishToTopicAsync(string topicArn, string messageContents, SubscribedQueueType subscribedQueueType, CancellationToken cancellationToken);
```

Also, I can't think of anywhere else in the code base where the `Async` suffix is used in methods that we write. I think the official MS naming recommends it but I'd only use it where there is a sync + async version of the same method, otherwise it becomes bloat as most methods are async.

Reply:
fixed in #[b804d5f](https://github.com/dlcs/protagonist/pull/568/commits/b804d5f01951e4c464698c50c473314754c82dad)